### PR TITLE
[Swaps] Add benchmarks for pm-pool operations

### PR DESCRIPTION
**Pull request by Chralt98**

### CODE REVIEW

File: zrml/swaps/src/lib.rs [hunk 1]
```diff
@@ -1850,17 +1855,23 @@ mod pallet {
...
+                    };
+                    ensure!(
+                        pool.pool_status == PoolStatus::Initialized,
+                        Error::<T>::InvalidStateTransition
+                    );
+                    pool.pool_status = PoolStatus::Active;
+                    Ok(pool.assets.len() as u32)
+                })?;
             Self::deposit_event(Event::PoolActive(pool_id));
             // TODO(#603): Fix weight calculation!
```
Comment (Chralt98) [inline zrml/swaps/src/lib.rs:1873]:
```suggestion
```

File: zrml/swaps/src/lib.rs [hunk 2]
```diff
@@ -1850,17 +1855,22 @@ mod pallet {
...
+                    } else {
+                        return Err(Error::<T>::PoolDoesNotExist.into());
+                    };
+                    ensure!(
+                        pool.pool_status == PoolStatus::Initialized,
+                        Error::<T>::InvalidStateTransition
+                    );
+                    pool.pool_status = PoolStatus::Active;
+                    Ok(pool.assets.len() as u32)
+                })?;
```
Comment (maltekliemann) [inline zrml/swaps/src/lib.rs:1871]: [nitpick] Too bad that you have to copy the `mutate_pool` code, but I don't see any other way. Suppose you did this...

```rust
Self::mutate_pool(...);
let pool = Pools::<T>::get(pool_id);
Ok(pool.assets.len() as u32)
```

Would that result in an additional read after substrate does its magic optimization?

**Comment by maltekliemann** [comment]
Starting my review now...

File: zrml/swaps/src/benchmarks.rs [hunk 3]
```diff
@@ -592,6 +608,69 @@ benchmarks! {
...
+    } verify {
+        let pool = Pallet::<T>::pool_by_id(pool_id).unwrap();
+        assert_eq!(pool.pool_status, PoolStatus::Closed);
+    }
+
+    destroy_pool {
+        let a in 2..T::MaxAssets::get().into();
+
+        let caller: T::AccountId = whitelisted_caller();
+        let (pool_id, assets, ..) = bench_create_pool::<T>(
```
Comment (maltekliemann) [inline zrml/swaps/src/benchmarks.rs:657]:
```suggestion
        let (pool_id, ..) = bench_create_pool::<T>(
```

File: zrml/swaps/src/benchmarks.rs [hunk 4]
```diff
@@ -592,6 +608,69 @@ benchmarks! {
...
+    } verify {
+        let pool = Pallet::<T>::pool_by_id(pool_id).unwrap();
+        assert_eq!(pool.pool_status, PoolStatus::Active);
+    }
+
+    close_pool {
+        let a in 2..T::MaxAssets::get().into();
+
+        let caller: T::AccountId = whitelisted_caller();
+        let (pool_id, assets, ..) = bench_create_pool::<T>(
```
Comment (maltekliemann) [inline zrml/swaps/src/benchmarks.rs:636]:
```suggestion
        let (pool_id, ..) = bench_create_pool::<T>(
```

**Review by maltekliemann — DISMISSED** [review]

**Review by vivekvpandya — APPROVED** [review]

File: zrml/swaps/src/lib.rs (hunk 2 above)
Comment (Chralt98) [reply zrml/swaps/src/lib.rs]: 34aa62a6765e7eafc8e93cefecba72f7a28dfb28

**Review by maltekliemann — APPROVED** [review]

**Review by vivekvpandya — APPROVED** [review]

**Pull request merged**